doped/generation.py

```
@@ -295,17 +289,31 @@ def get_neighbour_distances_and_symbols(
         )
         neighbour_tuples = [  # prune site_distances to remove any with distances within 0.02 Å of the
-            # previous n:
+            # previous n (and the same element as the previous n):
             neighbour_tuples[i]
             for i in range(len(neighbour_tuples))
             if neighbour_tuples[i][0] > min_dist
-            and (
-                i == 0
-                or abs(neighbour_tuples[i][0] - neighbour_tuples[i - 1][0]) > 0.02
-                or neighbour_tuples[i][1] != neighbour_tuples[i - 1][1]
+            and i == 0
```

Ruff (0.11.9)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)